go/processor/floria/processor.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"fmt"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"github.com/holiman/uint256"
+)
+
+func init() {
+	tosca.RegisterProcessorFactory("floria", newProcessor)
+}
+
+func newProcessor(interpreter tosca.Interpreter) tosca.Processor {
+	if interpreter == nil {
+		panic("interpreter is required")
+	}
+
+	return &processor{
+		interpreter: interpreter,
+	}
+}
+
+type processor struct {
+	interpreter tosca.Interpreter
+}
+
+func (p *processor) Run(
+	blockParams tosca.BlockParameters,
+	transaction tosca.Transaction,
+	context tosca.TransactionContext,
+) (tosca.Receipt, error) {
+
+	errorReceipt := tosca.Receipt{
+		Success: false,
+		GasUsed: transaction.GasLimit,
+	}
+
+	if err := buyGas(transaction, context); err != nil {
+		return errorReceipt, nil
+	}
+
+	if err := handleNonce(transaction, context); err != nil {
+		return errorReceipt, nil
+	}
+
+	if err := transferValue(transaction, context); err != nil {
+		return errorReceipt, nil
+	}
+
+	return tosca.Receipt{
+		Success:         true,
+		GasUsed:         transaction.GasLimit,
+		ContractAddress: nil,
+		Output:          []byte{},
+		Logs:            nil,
+	}, nil
+}
+
+func handleNonce(transaction tosca.Transaction, context tosca.TransactionContext) error {
+	stateNonce := context.GetNonce(transaction.Sender)
+	messageNonce := transaction.Nonce
+	if messageNonce != stateNonce {
+		return fmt.Errorf("nonce mismatch: %v != %v", messageNonce, stateNonce)
+	}
+
+	// Increment nonce
+	context.SetNonce(tosca.Address(transaction.Sender), context.GetNonce(tosca.Address(transaction.Sender))+1)
+
+	return nil
+}
+
+func buyGas(transaction tosca.Transaction, context tosca.TransactionContext) error {
+	gasPrice := transaction.GasPrice.ToUint256()
+	gas := uint256.NewInt(uint64(transaction.GasLimit))
+	gas = gas.Mul(gas, gasPrice)
+
+	// Buy gas
+	senderBalance := context.GetBalance(transaction.Sender).ToUint256()
+	if senderBalance.Cmp(gas) < 0 {
+		return fmt.Errorf("insufficient balance: %v < %v", senderBalance, gas)
```

Comment:
should we already setup an `errors.go` this could be a simple call to `errInsufficientBalance(senderBalance,gas)`

Reply:
As discussed in the standup, these errors are not going to be returned and are only used internal for this processor implementation.